### ISSUE TITLE
[java] JUnit tests should include assert() or fail() - false positives with lambdas

### ISSUE THREAD

**Issue by mwisnicki** [pmd/pmd#4655]
**Affects PMD Version:**

6.55.0
7.0.0

**Rule:**

[JUnitTestsShouldIncludeAssert](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)

**Description:**

**Code Sample demonstrating the issue:**

```java
@Test
void test() {
  Awaitility.await().atMost(Duration.ofSeconds(5)).untilAsserted(() -> {
    assertTrue(1 == 1);
  });
}
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *Codacy*

**Comment by jsotuyod** [comment]
This is a duplicate of #4272
